airflow/hooks/webhdfs_hook.py

```diff
@@ -0,0 +1,81 @@
+from airflow.hooks.base_hook import BaseHook
+from airflow.configuration import conf
+import logging
+
+from hdfs import InsecureClient, HdfsError
+from airflow.utils import AirflowException
+
+
+class WebHDFSHookException(AirflowException):
+    pass
+
+
+class WebHDFSHook(BaseHook):
+    """
+    Interact with HDFS. This class is a wrapper around the hdfscli library.
+    """
+    def __init__(self, webhdfs_conn_id='webhdfs_default'):
+        self.webhdfs_conn_id = webhdfs_conn_id
+
+    def get_conn(self):
+        """
+        Returns a hdfscli InsecureClient object.
+        """
+        nn_connections = self.get_connections(self.webhdfs_conn_id)
+        for nn in nn_connections:
+            try:
+                logging.debug('Trying namenode {}'.format(nn.host))
+                client = InsecureClient('http://{nn.host}:{nn.port}'.format(nn=nn))
```

Comment:
@artwr I am, I am! I think you can use KerberosClient instead of TokenClient (as we are not getting a delegation token for a user at the moment) with if configuration.conf.get("security") = "kerberos"?